app/src/main/java/com/example/criminalintent/presentation/crime_list/CrimeListFragment.kt

```diff
@@ -51,16 +53,31 @@ class CrimeListFragment : Fragment() {
         observeData()
     }
 
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+        inflater.inflate(R.menu.fragment_crime_list, menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.new_crime -> {
+                val crime = Crime()
```

Comment:
Think of what got you here. The user tapped a small icon in the app bar.
The user probably doesn't know what the icon means and tapped it to see what it does.
You now force the creation of a new crime.

That is bad UX.

My proposal to fix this:
- Just open the other screen for inputting the crime details from here
- Leave the logic of creating the crime to that screen
- This screen only displays the list
- The creation screen has a button either in the app bar called save, or with an icon OR a button on the bottom of the screen taking the entire width and centered called save. When pressed, the crime is inserted into the database, then the screen is exited
- If the user backs out of the creation screen without saving, a dialog is shown explaining to the user that their changes will not be saved. The dialog has options: continue editing - dismisses the dialog, discard changes - exits the screen without making any changes to the db